### PR TITLE
fix #249541 - Improvements to Plugin Manager/Creator

### PR DESCRIPTION
**Pull request by rjmatthews62**
All 3.0dev changes ported into 2.2 branch.

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Now with #3289 merged this needs a rebase

**Comment by lasconic** [comment]
Feel free to rebase this one too.

```
git checkout 2.2
git fetch upstream
git merge upstream/2.2
git checkout plugin_fix_2.2
git rebase 2.2
// fix the conflicts if any
git push -f upstream 2.2
```

File: mscore/qmlplugin.h [hunk 1]
```diff
@@ -36,10 +36,11 @@ extern int updateVersion();
...
 //   @P filePath             QString           source file path, without the file name (read only)
-//   @P version              QString
-//   @P description          QString
-//   @P pluginType           QString
-//   @P dockArea             QString
+//   @P version              QString           version of this plugin
+//   @P description          QString           human readable description, displayed in Plugin Manager
+//   @P pluginType           QString           type may be dialog, dock, or not defined.
+//   @P dockArea             QString           where to dock on main screen. left,top,bottom, right(default)
+//   @P requiresScore        bool              whether the plugin requires an existing score to run
```
Comment (lasconic) [inline mscore/qmlplugin.h]: requireScores doesn't exist in 2.2

**Comment by rjmatthews62** [comment]
Also done, I believe )including requiresScore

**Pull request merged**